crates/rooch/src/commands/account/commands/update.rs

```diff
@@ -0,0 +1,140 @@
+    #[clap(short = 'a', long = "address")]
+    address: String,
+    #[clap(short = 'm', long = "mnemonic-phrase")]
+    mnemonic_phrase: String,
+    #[clap(flatten)]
+    pub context_options: WalletContextOptions,
+    /// Command line input of crypto schemes (ed25519, multied25519, ecdsa, or schnorr)
+    #[clap(short = 's', long = "scheme", arg_enum)]
+    pub crypto_schemes: BuiltinScheme,
+}
+
+#[async_trait]
+impl CommandAction<ExecuteTransactionResponseView> for UpdateCommand {
+    async fn execute(self) -> RoochResult<ExecuteTransactionResponseView> {
+        println!("{:?}", self.mnemonic_phrase);
+
+        let mut context = self.context_options.build().await?;
+
+        match BuiltinScheme::from_flag_byte(self.crypto_schemes.flag()) {
+            Ok(scheme) => {
+                let existing_address =
+                    RoochAddress::from_str(self.address.as_str()).map_err(|e| {
+                        RoochError::CommandArgumentError(format!(
+                            "Invalid Rooch address String: {}",
+                            e
+                        ))
+                    })?;
+
+                let public_key: PublicKey = context
+                    .config
+                    .keystore
+                    .update_address_with_key_pair_from_scheme(
+                        &existing_address,
+                        self.mnemonic_phrase,
+                        scheme,
+                        None,
+                    )
+                    .map_err(|e| RoochError::UpdateAccountError(e.to_string()))?;
+
+                println!(
+                    "{}",
+                    AccountAddress::from(existing_address).to_hex_literal()
+                );
+                println!(
+                    "Generated a new keypair for an existing address on scheme {:?} [{existing_address}]",
+                    scheme.to_owned()
+                );
+
+                let (address, module_name) = match scheme {
+                    BuiltinScheme::Ed25519 => (
+                        Ed25519Validator::MODULE_ADDRESS,
+                        Ed25519Validator::MODULE_NAME,
+                    ),
+                    BuiltinScheme::MultiEd25519 => todo!(),
+                    BuiltinScheme::Ecdsa => (
+                        EcdsaK1Validator::MODULE_ADDRESS,
+                        EcdsaK1Validator::MODULE_NAME,
+                    ),
+                    BuiltinScheme::EcdsaRecoverable => (
+                        EcdsaK1RecoverableValidator::MODULE_ADDRESS,
+                        EcdsaK1RecoverableValidator::MODULE_NAME,
+                    ),
+                    BuiltinScheme::Schnorr => (
+                        SchnorrValidator::MODULE_ADDRESS,
+                        SchnorrValidator::MODULE_NAME,
+                    ),
+                };
+
+                let validator_struct_arg: Box<StructTag> =
+                    scheme.create_validator_struct_tag(address, module_name.to_string())?;
```

Comment:
I've updated it. Now it looks more elegant. However, due to cyclic import dependencies between rooch-types and rooch-framework, I couldn't refactor Validators' address and module into BuiltinScheme.